xlunch.c

```diff
@@ -1031,10 +1031,14 @@ void reset_prompt()
 
 void run_command(char * cmd_orig)
 {
-
-    char cmd[512];
+    char *cmd;
     char *array[100] = {0};
-    strcpy(cmd,cmd_orig);
+
+    cmd = strdup(cmd_orig);
+    if (!cmd) {
+        fprintf(stderr, "Out of memory!\n");
+        exit(EXIT_FAILURE);
```

Comment:
xlunch has it's own list of exit codes defined at the start of the file. I think this one should be an `ALLOCERROR` from that list.

Reply:
Oh, I didn't notice that. Sorry. I've fixed it and force pushed a new version.

Reply:
No worries, it's an easy one to miss, but the categorised error/return codes are nice for scripting against xlunch.